src/ucx_plugin.c

```diff
@@ -685,152 +716,178 @@ ncclResult_t nccl_ucx_isend(void *send_comm, void *data, int size, int tag, void
     }
   }
 
-#if UCP_API_VERSION >= UCP_VERSION(1, 10)
-  ucp_request_param_t params;
+  req = ucx_request_get(comm);
+  if (req == NULL) {
+    return ncclInternalError;
+  }
```

Comment:
maybe for send flow we don't need this since there is always a single buffer to send

Reply:
let me know if this is what you meant but i think we need it as-is because:
- we want to return `*request = req` with always same type as other callbacks
- api does not seem to guarantee that we have only one req instance in-flight, so we still need to allocate one
- our callback `->test()` that does progress will need to be called with a valid request